src/Container.php

```diff
@@ -469,11 +475,19 @@ public static function create(): self
                     $container->getConfiguration()->noProgress()
                 );
             },
+            MutationTestingResultsCollectorSubscriberFactory::class => static function (self $container): MutationTestingResultsCollectorSubscriberFactory {
+                return new MutationTestingResultsCollectorSubscriberFactory(
+                    $container->getMetricsCalculator(),
+                    // TODO ResultsCollector can have multiple strategies on results accumulations, e.g. whenever uncovered results need saving
+                    $container->getResultsCollector()
```

Comment:
Here's the place to inject something, that will stand between `ResultsCollectorSubscriber` and actual `ResultsCollector` doing the filtering of unneeded results.